### PR TITLE
Fix destructor declaration indentation

### PR DESCRIPTION
**Pull request by CaseyCarter**
..and apparently some trailing whitespace cleanup because my text editor hates it as much as I do.

### CODE REVIEW

File: source/iostreams.tex [hunk 1]
```diff
@@ -10420,7 +10420,7 @@
       path(const Source& source, const locale& loc);
     template <class InputIterator>
       path(InputIterator first, InputIterator last, const locale& loc);
-   ~path();
+    ~path();
```
Comment (CaseyCarter) [inline source/iostreams.tex]: Actual change is here.

**Comment by jwakely** [comment]
I think this is intentional, so the destructor lines up with the constructor, and we do it in a few places:

containers.tex:   ~vector();                                                                                                                                                                                      
iostreams.tex:   ~Init();                                                                                                                                                                                         
iostreams.tex:   ~path();                                                                                                                                                                                         
iostreams.tex:   ~file_status();                                                                                                                                                                                  
iostreams.tex:   ~directory_entry();                                                                                                                                                                              
iostreams.tex:   ~directory_iterator();                                                                                                                                                                           
iostreams.tex:   ~recursive_directory_iterator();                                                                                                                                                                 
iterators.tex:   ~istream_iterator() = default;                                                                                                                                                                   
iterators.tex:   ~ostream_iterator();                                                                                                                                                                             
locales.tex:   ~numpunct_byname();                                                                                                                                                                                
numerics.tex:   ~mask_array();                                                                                                                                                                                    
numerics.tex:   ~indirect_array();                                                                                                                                                                                
strings.tex:   ~basic_string();                                                                                                                                                                                   
utilities.tex:   ~allocator();      

That doesn't mean we have to keep it though.

**Comment by CaseyCarter** [comment]
>  we do it in a few places

Blech, I only looked at a couple which of course didn't use this device. It would be nice to pick a rule and be consistent throughout the standard. (See `template<` vs. `template <`)

I'll repush with them all fixed. (Yes, I said "fixed." I'm totally unbiased.)

**Comment by CaseyCarter** [comment]
> containers.tex: ~vector();

The fact that `vector<bool>` uses this device clearly indicates that it's very, very wrong.

**Comment by CaseyCarter** [comment]
There's a consistent fix throughout.

File: source/containers.tex [hunk 2]
```diff
@@ -4155,7 +4155,7 @@
 \begin{itemdescr}
 \pnum
 \effects If \tcode{sz < distance(begin(), end())}, erases the last \tcode{distance(begin(),
-end()) - sz} elements from the list. Otherwise, inserts \tcode{sz - distance(begin(), end())}
```
Comment (tkoeppe) [inline source/containers.tex]: Could you please submit the trailing whitespace deletions as a separate pull request?

Comment (jensmaurer) [inline source/containers.tex]: @tkoeppe: See #1468.

Comment (tkoeppe) [inline source/containers.tex]: @CaseyCarter: Please rebase.

File: source/iostreams.tex (hunk 1 above)
Comment (CaseyCarter) [inline source/iostreams.tex]: ...and everywhere else.

File: source/containers.tex (hunk 2 above)
Comment (CaseyCarter) [inline source/containers.tex]: Done.
